Website build running into rate limitings

Mostly after releasing a new version for one of our ReactPHP components, we need to trigger the website build in order to show each new released version there. When running the GitHub actions deployment workflow for this, we recently tend to run into rate limitings:

```
Temporary error, will retry in 60s: API rate limit exceeded for installation ID 14316185.
PHP Fatal error:  Uncaught Github\Exception\RuntimeException: API rate limit exceeded for installation ID 14316185. in /home/runner/work/website/website/vendor/knplabs/github-api/lib/Github/HttpClient/Plugin/GithubExceptionThrower.php:126
Stack trace:
#0 /home/runner/work/website/website/vendor/php-http/httplug/src/Promise/HttpFulfilledPromise.php(34): Github\HttpClient\Plugin\GithubExceptionThrower->Github\HttpClient\Plugin\{closure}(Object(GuzzleHttp\Psr7\Response))
#1 /home/runner/work/website/website/vendor/knplabs/github-api/lib/Github/HttpClient/Plugin/GithubExceptionThrower.php(127): Http\Client\Promise\HttpFulfilledPromise->then(Object(Closure))
#2 /home/runner/work/website/website/vendor/php-http/client-common/src/Plugin/VersionBridgePlugin.php(19): Github\HttpClient\Plugin\GithubExceptionThrower->doHandleRequest(Object(GuzzleHttp\Psr7\Request), Object(Closure), Object(Closure))
#3 /home/runner/work/website/website/vendor/php-http/client-common/src/PluginClient.php(161): Github\HttpClient\Plugin\GithubExceptionThrower in /home/runner/work/website/website/vendor/knplabs/github-api/lib/Github/HttpClient/Plugin/GithubExceptionThrower.php on line 126
Error: Process completed with exit code 255.
```
It sometimes works when retrying to run the failed workflow, but with each release it seems to get harder for this deployment process to be successful. I didn't investigate this in detail yet, so I'm happy about your input on this.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -7,15 +7,15 @@ on:
 
 jobs:
   Deploy:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
       - uses: shivammathur/setup-php@v2
         with:
           php-version: 7.1
       - run: composer install
       - run: mkdir ~/.ssh && echo "${{ secrets.DEPLOY_KEY }}" > ~/.ssh/id_rsa && chmod 400 ~/.ssh/id_rsa
-      - run: echo 'GITHUB_TOKEN="${{ secrets.GITHUB_TOKEN }}"' > .env && cat .env.dist >> .env
+      - run: echo 'GITHUB_TOKEN="${{ secrets.PAT || secrets.GITHUB_TOKEN }}"' > .env && cat .env.dist >> .env
       - run: git config --global user.name "GitHub Actions" && git config --global user.email "actions@github.com"
       - run: bin/build
       - run: bin/build --deploy --no-component-update

--- a/README.md
+++ b/README.md
@@ -1,7 +1,7 @@
 # Website
 
 [![CI status](https://github.com/reactphp/website/actions/workflows/ci.yml/badge.svg)](https://github.com/reactphp/website/actions)
-![Last deployed](https://img.shields.io/github/last-commit/reactphp/reactphp.github.io?label=last%20deployed&logo=github)
+[![Last deployed](https://img.shields.io/github/last-commit/reactphp/reactphp.github.io?label=last%20deployed&logo=github)](https://github.com/reactphp/reactphp.github.io)
 
 Source code of reactphp.org.
 
@@ -79,6 +79,10 @@ deployment script (see previous chapter).
 > Make sure the required `DEPLOY_KEY` secret is set in the repository settings on GitHub.
 > See [action documentation](https://github.com/JamesIves/github-pages-deploy-action#using-an-ssh-deploy-key-)
 > for more details.
+> On top of this, you're recommended to add a [personal access token](https://github.com/settings/tokens)
+> as a repository secret with the name `PAT` to avoid running into secondary rate limits.
+> If this secret is not found, it will fall back to the automatic `GITHUB_TOKEN`
+> secret, which may cause the build to fail occasionally.
 
 ## License
 

